Document touch() doesn't clear the stat cache.

Crell opened the pull request:
Thanks to @derickr for pointing this out to me.  It drove me batty for days. :smile:

TimWolla reviewed (changes requested):
This is not unique about `touch()`. None of the file operations clear the stat cache:

```
<?php
var_dump(filemtime('/tmp/foo.txt'), filesize('/tmp/foo.txt'));
file_put_contents('/tmp/foo.txt', str_repeat('x', random_int(0, 999)));
clearstatcache();
var_dump(filemtime('/tmp/foo.txt'), filesize('/tmp/foo.txt'));
```

Crell commented:
Seems like we should add the same warning to other functions, then.  Is it worth making an entity out of it, or just copy-paste-tweak?

Girgias commented:
We already have the `note.clearstatcache` entity. Is that insufficient?

Girgias commented on reference/filesystem/functions/touch.xml:
```diff
@@ -135,6 +135,27 @@ if (!touch('some_file.txt', $time)) {
+ </refsect1>
+
+ <refsect1 role="seealso">
+  &reftitle.seealso;
+  <para>
+   <simplelist>
+    <member><function>clearstatcache</function></member>
+    <member><function>fileatime</function></member>
+    <member><function>filemtime</function></member>
+   </simplelist>
+  </para>
```
```suggestion
  &reftitle.seealso;
  <simplelist>
   <member><function>clearstatcache</function></member>
   <member><function>fileatime</function></member>
   <member><function>filemtime</function></member>
  </simplelist>
```

Crell commented:
What is insufficient is my awareness that entity already exists. :smile: Switched to using that, and added `touch` to the list on the `clearstatcache` page, which was missing.

TimWolla commented on reference/filesystem/functions/clearstatcache.xml:
```diff
@@ -57,8 +57,9 @@
-   <function>fileperms</function>.
+   <function>filetype</function>,
+   <function>fileperms</function>, and.
+   <function>touch</function>.
```
This change is wrong. This list lists the functions that _read_ from the stat cache. `touch()` is a _setter_.

Crell replied:
The description says nothing about read/  It just says "Affected."

TimWolla replied:
The context is:

> When you use stat(), lstat(), or any of the other functions listed in the affected functions list (below), PHP caches the information those functions return in order to provide faster performance.

`touch()` is not an affected function, because it does not interact with the stat cache at all. It does not (need to) know about it.

TimWolla commented on reference/filesystem/functions/touch.xml:
```diff
@@ -135,6 +135,16 @@ if (!touch('some_file.txt', $time)) {
  <refsect1 role="notes">
   &reftitle.notes;
   &note.filesystem-time-res;
+  &note.clearstatcache;
```
And likewise is this change wrong.

Crell replied:
Then do we need a new entity for statcache impacting write functions?  Because the root issue is I didn't realize touch didn't reset the cache, so couldn't understand why my touch() call didn't seem to do anything.  So I want *something* on the touch page to indicate that.  You and @Girgias figure out what you want there, I've already made two proposals and both of them someone has objected to.

TimWolla replied:
> Then do we need a new entity for statcache impacting write functions?

None of them impact the stat*cache* (except for `unlink()`, which automatically clears it according to the manual). But virtually all of them impact the stat*data*, because writes implicitly affect the mtime (and usually also the file size).

Note that `exec()` and friends might also affect the stat*data*, because you could `exec('touch foo.txt')`. `Dom\HTMLDocument::saveHtml()` does as well.

I do not see a value-add in adding a reference to the stat*cache* to every function that possibly affects the stat*data* and as far as I can tell all the functions reading from the statcache have the note already.

TimWolla commented on reference/filesystem/functions/touch.xml:
```diff
@@ -135,6 +135,16 @@ if (!touch('some_file.txt', $time)) {
+ </refsect1>
+
+ <refsect1 role="seealso">
+  &reftitle.seealso;
```
This should probably list `filemtime` first, because the mtime is the relevant field that is modified by `touch`.

Crell replied:
touch can modify both properties, so I listed them in alphabetical order.

lacatoire commented:
Closing as obsolete
